pkg/kudoctl/packages/types.go

```diff
@@ -27,25 +28,60 @@ type Resources struct {
 	Instance        *v1beta1.Instance
 }
 
-type Parameter []v1beta1.Parameter
+// Modified v1beta1.Parameter that allows for defaults provided as YAML.
+type Parameter struct {
+	DisplayName string                `json:"displayName,omitempty"`
+	Name        string                `json:"name,omitempty"`
+	Description string                `json:"description,omitempty"`
+	Required    *bool                 `json:"required,omitempty"`
+	Default     interface{}           `json:"default,omitempty"`
+	Trigger     string                `json:"trigger,omitempty"`
+	Type        v1beta1.ParameterType `json:"type,omitempty"`
+}
 
-// Templates is a map of file names and stringified files in the template folder of an operator
-type Templates map[string]string
+type Parameters []Parameter
 
 // Len returns the number of params.
 // This is needed to allow sorting of params.
-func (p Parameter) Len() int { return len(p) }
+func (p Parameters) Len() int { return len(p) }
 
 // Swap swaps the position of two items in the params slice.
 // This is needed to allow sorting of params.
-func (p Parameter) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+func (p Parameters) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
 
 // Less returns true if the name of a param a is less than the name of param b.
 // This is needed to allow sorting of params.
-func (p Parameter) Less(x, y int) bool {
+func (p Parameters) Less(x, y int) bool {
 	return p[x].Name < p[y].Name
 }
 
+// ToAPI converts parameters to an array of 'Parameter' defined in the KUDO API.
+func (p Parameters) ToAPI() ([]v1beta1.Parameter, error) {
```

Comment:
I'm not sure what to call this function... but `toAPI` is strange... I was hoping the godoc would make it clear and it doesn't.  The reader is forced to dive into the method and read the code to understand.   Is there a pattern here you are striving for?  do you expect `ToAPI` on other types?    Not a blocker... just doesn't seem readable (from the calling point stand point)

Reply:
This could become a pattern if our operator package types divert further from the types used in the CRDs.
How about `ToCRDType`?